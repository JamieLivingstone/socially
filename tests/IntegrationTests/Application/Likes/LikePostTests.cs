using System.Linq;
using System.Threading.Tasks;
using Application.Common.Exceptions;
using Application.Likes.Commands.LikePost;
using Domain.Entities;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Likes;

[TestFixture]
public class LikePostTests : TestBase
{
  [Test]
  public void GivenPostDoesNotExist_ThrowsNotFoundException()
  {
    var command = new LikePostCommand
    {
      Slug = "does-not-exist"
    };

    async Task Handler() => await SendAsync(command);

    Assert.ThrowsAsync(typeof(NotFoundException), Handler);
  }

  [Test]
  public async Task GivenAValidRequest_LikesPost()
  {
    var target = Seed.Posts().First();

    await SendAsync(new LikePostCommand
    {
      Slug = target.Slug
    });

    Snapshot.Match(await FindAsync<Like>(l => l.ObserverId == Seed.CurrentUserId && l.PostId == target.Id));
  }
}
