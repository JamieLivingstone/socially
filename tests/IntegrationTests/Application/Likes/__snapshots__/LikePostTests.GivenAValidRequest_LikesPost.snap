{
  "ObserverId": 1,
  "Observer": null,
  "PostId": 1,
  "Post": null
}
